src/java/org/apache/cassandra/utils/FastByteOperations.java

```diff
@@ -262,7 +321,7 @@ public void copy(byte[] src, int srcPosition, ByteBuffer trg, int trgPosition, i
             if (trg.hasArray())
                 System.arraycopy(src, srcPosition, trg.array(), trg.arrayOffset() + trgPosition, length);
             else
-                copy(null, srcPosition + theUnsafe.getLong(src, Unsafe.ARRAY_BYTE_BASE_OFFSET), trg, trgPosition, length);
+                copy(src, (long) srcPosition + Unsafe.ARRAY_BYTE_BASE_OFFSET, trg, trgPosition, length);
```

Comment:
Just to make it extra visible which method is called, I would cast `src` to `Object` as well.